auth/auth-router.js

```diff
@@ -1,11 +1,69 @@
-const router = require('express').Router();
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-undef */
 
-router.post('/register', (req, res) => {
-  // implement registration
+// Imports
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+// Model
+const Users = require("./users-model");
+
+function makeToken(user) {
+  const payload = {
+    sub: user.id,
+    username: user.username
+  };
+  const options = {
+    expiresIn: "1d"
+  };
+  const token = jwt.sign(
+    payload,
+    process.env.JWT_SECRET || "thesecret",
+    options
+  );
+  return token;
+}
+// CRUD
+router.post("/register", (req, res) => {
+  let user = req.body;
+  const hash = bcrypt.hashSync(user.password, 10);
+  user.password = hash;
+
+  Users.add(user)
+    .then(saved => {
+      res.status(201).json(saved);
```

Comment:
This will return the credentials of the new user to the client including the hashed password.
The hashed password should not be revealed and should be kept secret at all times